backend/config.go

```diff
@@ -119,6 +168,18 @@ type ReplicaInfo struct {
 
 // Connect opens connections to the replicas in the configuration.
 func (cfg *Config) Connect(replicas []ReplicaInfo) (err error) {
+	opts := *cfg.optsPtr
+	cfg.optsPtr = nil // we don't need to keep the options around beyond this point, so we'll allow them to be GCed.
```

Comment:
Is this an important point? I mean is this causing memory leaks or something? Maybe it would be better to make a copy of options and let that be GCed when exiting the method. I mean `Connect` should be called rarely, right?